### DISCUSSION TITLE
ModuleNotFoundError: No module named 'solcx'

### DISCUSSION THREAD

**Discussion by weilainicolas** [smartcontractkit/full-blockchain-solidity-course-py#510] [Q&A] [answered]
I have ran pip3 install py-solc-x --user to install py-solc-x 
however when I run python deploy.py still got this error message as in the title. 

`from solcx import compile_standard, install_solc
import json

import os

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compile_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

print(compile_sol)`

can anyone help?

**Comment by weilainicolas** [comment, marked as answer]
I uninstalled and re-installed py-solc-x, and it worked.

**Reply by cromewar** [reply]
Sorry for the delay answering, I'm glad you solved your problem.

**Reply by ghost** [reply]
how do u uninustall py-solc-x ?

**Reply by weilainicolas** [reply]
I can find it in my terminal but not in vs code. I guess it was an environment issue. So I uninstalled in from local terminal. 
I am a beginner, so dealing with different environment is very challenge for me. I met more problems later on. The best solution I found is to reboot a windows machine with linux. and install multiple linux machines on that computer one for each project. When 1 project messed up, I just reboot it and start over.

**Reply by shucks18** [reply]
yo wdym by that?

**Reply by mr-ziash** [reply]
> I uninstalled and re-installed py-solc-x, and it worked.

how do you uninstall?

**Reply by cromewar** [reply]
@mr-ziash use `pip uninstall`

**Reply by mr-ziash** [reply]
> @mr-ziash use `pip uninstall`
pip uninstall solcx doesn't work i don't know why

**Reply by cromewar** [reply]
The package name is py-solc-x

**Reply by mr-ziash** [reply]
> The package name is py-solc-x

thank you I uninstalled and reinstalled but i'm still getting the same error aka this one: ModuleNotFoundError: No module named 'solcx', when I run the code or I get INFO: Could not find files for the given pattern(s), can you help, please? would really appreciate it

**Reply by cromewar** [reply]
tou should select the correct python interpreter on vscode (bottom left corner) and INFO: Could not find files for the given pattern(s) is not an error.

**Reply by mr-ziash** [reply]
> 

tried changing the interpreter but still nothing

**Reply by mr-ziash** [reply]
> tou should select the correct python interpreter on vscode (bottom left corner) and INFO: Could not find files for the given pattern(s) is not an error.

hey man thanks for the help, i found this solution the pip version (you type pip --version) in the terminal should be the same as the python version (python --version) and the python version in the bottom left corner, but I'm still getting that INFO: could not find files for the gives pattern(s), is that okay? also I don't see that bunch of code that was shown in the video I just get that info thing

**Reply by weilainicolas** [reply]
I guess this may be related to your PATH set up. Because your pip version is different, the package is installed in a different path. Can you try to install it using a correct pip version?

**Reply by cromewar** [reply]
Does @weilainicolas suggestion worked for you @mr-ziash ?

**Reply by mr-ziash** [reply]
> 

I added the pip path in the script folder to my PATH settings but i'm still getting INFO: Could not find files for the given pattern(s)., will this prevent the code from running properly? @cromewar @weilainicolas

**Reply by cromewar** [reply]
NFO: Could not find files for the given pattern(s) this is not an error, is just something related on the translation between instructions
Powershell to CMD con Windows
bash to zsh on Mac or linux

But you have nothing to worry about @mr-ziash

**Comment by spencergoff** [comment]
I resolved this issue by using the command python3 deploy.py instead of python deploy.py .

**Comment by weilainicolas** [comment]
Good job! Thanks for sharing.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
